### PR TITLE
Rename Workflow to WorkflowEntrypoint

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
`Workflow` conflicts with the binding class which is also called `Workflow`

Since we have `WorkerEntrypoint`, `WorkflowEntrypoint` feels like a much more appropriate and descriptive name of the base class

This is of course a breaking change, but these are all scoped under `experimental` for now so that is okay

### CODE REVIEW

**Comment by sidharthachatterjee** [comment]
I think this doesn't require an internal PR anymore because I'm only seeing references to workerd submodule but I'd like one of us to confirm

**Review by LuisDuarte1 — APPROVED** [review]

**Comment by fhanau** [comment]
> I think this doesn't require an internal PR anymore because I'm only seeing references to workerd submodule but I'd like one of us to confirm

Can you elaborate what you mean here? For the downstream project, workerd is being used the same way as before (as a submodule) so there should be an internal PR for this.

**Comment by sidharthachatterjee** [comment]
> > I think this doesn't require an internal PR anymore because I'm only seeing references to workerd submodule but I'd like one of us to confirm
> 
> Can you elaborate what you mean here? For the downstream project, workerd is being used the same way as before (as a submodule) so there should be an internal PR for this.

The file that I saw references to `entrypointClasses` doesn't seem to exist anymore in the downstream project

**Comment by sidharthachatterjee** [comment]
Never mind, I was looking at the wrong base branch @fhanau Internal PR is now ready as well and waiting on CI

**Review by a-robinson — APPROVED** [review]

**Review by a-robinson — APPROVED** [review]
LGTM but will wait on internal CI before merging.

**Pull request merged**
